engine/util/src/main/scala/pl/touk/nussknacker/engine/util/typing/JsonToTypedMapConverter.scala

```diff
@@ -0,0 +1,29 @@
+package pl.touk.nussknacker.engine.util.typing
+
+import io.circe.{Json, JsonObject}
+import pl.touk.nussknacker.engine.api.typed.TypedMap
+import scala.collection.JavaConverters._
+
+object JsonToTypedMapConverter {
+
+  def jsonToTypedMap(json: Json): TypedMap =
+    jsonObjectToTypedMap(json.asObject.getOrElse(JsonObject.empty))
+
+  private def jsonToAny(json: Json): Any = {
+    json.fold(
+      jsonNull = null,
+      jsonBoolean = identity,
+      jsonNumber = d => java.math.BigDecimal.valueOf(d.toDouble), //prevent scientific notation
```

Comment:
I've added a test that will show that this implementation loose precision. Because of this bug: https://github.com/circe/circe/issues/1101 it will fail even if you change it, but after upgrade to: https://github.com/TouK/nussknacker/pull/2208/files it will work correctly. IMO we should handle this that way: `d.toBigDecimal.map(_.bigDecimal).getOrElse(d.toDouble)`. I'm not sure about `getOrElse` part - it handles the case when decimal overflows `BigDecimal`